### ISSUE TITLE
When switching proxies to TLS routing, terraform provider no longer connects

### ISSUE THREAD

**Issue by nickbrennan1** [gravitational/teleport#9466]
On a terraform managed cluster, if TLS routing is enabled on the cluster, the provider is no longer able to communicate with the cluster.

Instead, any new action (show, apply etc) reports an error:

```
Error: all connection methods failed
	failed to connect to addr example.teleport.com:443 as a reverse tunnel proxy
		context deadline exceeded, 
	failed to connect to addr example.teleport.com:443 as an auth server
		context deadline exceeded, 
	failed to connect to addr example.teleport.com:443 as a web proxy
		context deadline exceeded
```

It appears due to the provider not invoking `tsh` no local proxy is activated.

### Reproduction Steps
1. Create ha-cluster @ 8.0.0+
2. Install TF provider @ 8.0.5
3. Activate the terraform provider to manage the cluster, create role etc
4. Activate TLS routing on the cluster
5. Attempt to connect to the cluster

**Comment by pschisa** [comment]
This seems to be the same issue hit after Teleport cloud was upgraded to enable TLS routing with 8.0.7 Terraform provider:
~~~
Error: all connection methods failed failed to connect to addr example.teleport.sh:443 as a reverse tunnel proxy context deadline exceeded, failed to connect to addr example.teleport.sh:443 as a web proxy context deadline exceeded, failed to connect to addr example.teleport.sh:443 as an auth server context deadline exceeded
with provider["gravitational.com/teleport/teleport"]
on provider.tf line 61, in provider "teleport":
~~~

**Comment by r0mant** [comment]
@pschisa @nickbrennan1 This should have been fixed in https://github.com/gravitational/teleport-plugins/pull/406 and available starting from 8.0.8. Both versions mentioned in this ticket are older than 8.0.8. cc @smallinsky
